test/integration/sequelize.transaction.test.js

```diff
@@ -1,141 +1,181 @@
 'use strict';
 
+const sinon = require('sinon');
 const chai = require('chai'),
   expect = chai.expect,
   Support = require('./support'),
   Transaction = require('sequelize/lib/transaction'),
   current = Support.sequelize,
   delay = require('delay');
 
-if (current.dialect.supports.transactions) {
+const sequelize = Support.sequelize;
+const dialectName = sequelize.dialect.name;
 
-  describe(Support.getTestDialectTeaser('Sequelize#transaction'), () => {
+describe(Support.getTestDialectTeaser('Sequelize#transaction'), () => {
+  if (!current.dialect.supports.transactions) {
+    return;
+  }
 
-    describe('then', () => {
-      it('gets triggered once a transaction has been successfully committed', async function() {
-        let called = false;
+  const stubs = [];
+
+  afterEach(() => {
+    for (const stub of stubs) {
+      stub.restore();
+    }
+
+    stubs.length = 0;
+  });
+
+  describe('Transaction#commit', () => {
+    it('returns a promise that resolves once the transaction has been committed', async function() {
+      const t = await this
+        .sequelize
+        .transaction();
+
+      await expect(t.commit()).to.eventually.equal(undefined);
+    });
+
+    // we cannot close a sqlite connection, but there also cannot be a network error with sqlite.
+    // so this test is not necessary for that dialect.
+    if (dialectName !== 'sqlite') {
+      it('does not pollute the pool with broken connections if commit fails', async function() {
+        const initialPoolSize = this.sequelize.connectionManager.pool.size;
+
+        stubs.push(sinon.stub(this.sequelize.queryInterface, 'commitTransaction').rejects(new Error('Oh no, an error!')));
 
         const t = await this
           .sequelize
           .transaction();
 
-        await t.commit();
-        called = 1;
-        expect(called).to.be.ok;
+        await expect(t.commit()).to.be.rejectedWith('Oh no, an error!');
+
+        // connection should have been destroyed
+        expect(this.sequelize.connectionManager.pool.size).to.eq(Math.max(0, initialPoolSize - 1));
```

Comment:
I don't even understand my confusion and why this line bothers me but doesn't this somehow allow for initialPoolSize to be 0? Is that a realistic scenario?

Reply:
Hmm... Okay then? Maybe? 😂